src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -509,7 +511,15 @@ keyToAddress
 keyToAddress (Key xpub) =
     Address $ CBOR.toStrictByteString $ encodeAddress xpub encodeAttributes
   where
-    encodeAttributes = CBOR.encodeMapLen 0
+    encodeAttributes = case network of
+        Testnet -> mempty
+            <> CBOR.encodeMapLen 1
+            <> CBOR.encodeWord 2
+            <> CBOR.encodeBytes (
+                  CBOR.toStrictByteString
+                $ encodeProtocolMagic (protocolMagic Testnet))
```

Comment:
Little risky to bring into separate pr (?), as this change on its own is unmotivated / meant to fix tests that don't exist yet

Reply:
I wouldn't say "risky" but "wise" :) 
This is a valid change so, I'd expect test to either pass, or highlight a failure somewhere.